Description: Add protected method in PrivateModule for binding provision listeners.
Author: Ghislain Touratier <ghislain.touratier@gmail.com>
Bug-Google: http://code.google.com/p/google-guice/issues/detail?id=748
Last-Update: 2013-05-12

diff --git a/core/src/com/google/inject/PrivateModule.java b/core/src/com/google/inject/PrivateModule.java
index fcc7933..4f66e0a 100644
--- a/core/src/com/google/inject/PrivateModule.java
+++ b/core/src/com/google/inject/PrivateModule.java
@@ -24,6 +24,7 @@ import com.google.inject.binder.AnnotatedElementBuilder;
 import com.google.inject.binder.LinkedBindingBuilder;
 import com.google.inject.matcher.Matcher;
 import com.google.inject.spi.Message;
+import com.google.inject.spi.ProvisionListener;
 import com.google.inject.spi.TypeConverter;
 import com.google.inject.spi.TypeListener;
 
@@ -293,4 +294,12 @@ public abstract class PrivateModule implements Module {
       TypeListener listener) {
     binder().bindListener(typeMatcher, listener);
   }
+  
+  /**
+   * @see Binder#bindListener(com.google.inject.matcher.Matcher, com.google.inject.spi.ProvisionListener...)
+   */
+  protected void bindListener(Matcher<? super Binding<?>> bindingMatcher,
+      ProvisionListener... listeners) {
+    binder().bindListener(bindingMatcher, listeners);
+  }
 }
